### PR TITLE
BAN-1615: Implement new "My offers "

### PR DESCRIPTION
**Pull request by iamsphere**
## Screenshots

image

## Issue link

https://linear.app/banx-gg/issue/BAN-1615/fe-p2p-pools-implement-new-my-offers-ui

## Vercel preview

https://banx-ui-git-feature-ban-1615-frakt.vercel.app/

### CODE REVIEW

File: src/pages/OffersPage/components/OffersTabContent/components/ActiveLoansTable/ActiveLoansTable.module.less [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+.tableRoot table {
+  tr > td {
+    width: 160px;
+
+    &:first-child {
+      min-width: 176px;
+    }
+
+    &:nth-last-child(-n + 2) {
```
[the remark this answers is not in the document]
Comment (iamsphere) [reply src/pages/OffersPage/components/OffersTabContent/components/ActiveLoansTable/ActiveLoansTable.module.less:13]: Ну такая запись, чтобы были выбраны последние два дочерних элемента.

File: src/pages/OffersPage/hooks.ts [hunk 2]
```diff
@@ -3,41 +3,112 @@ import { useEffect, useMemo } from 'react'
...
+  addLoans: (loan, walletPublicKey) => {
+    if (!walletPublicKey) return
+
+    return set(
+      produce((state: OptimisticLenderLoansState) => {
+        state.loans.push(convertLoanToOptimistic(loan, walletPublicKey))
+      }),
+    )
+  },
+  findLoans: (loanPubkey, walletPublicKey) => {
```
[the remark this answers is not in the document]
Comment (iamsphere) [reply src/pages/OffersPage/hooks.ts]: [Rename findLoans => findLoan](https://github.com/frakt-solana/banx-ui/pull/381/commits/3d244316fecffe32cd4f9102179686a061cd6579)

File: src/pages/OffersPage/components/OffersTabContent/hooks/useOffersTabContent.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+  const { sortedData, sortParams } = useSortedData(filteredData)
+
+  const searchSelectOptions = uniqBy(
+    data.map(({ loans, collectionMeta }) => ({
+      collectionName: collectionMeta.collectionName,
+      collectionImage: collectionMeta.collectionImage,
+      claim: sumBy(loans, calculateClaimValue),
+    })),
+    ({ collectionName }) => collectionName,
+  )
```
[the remark this answers is not in the document]
Comment (iamsphere) [reply src/pages/OffersPage/components/OffersTabContent/hooks/useOffersTabContent.ts]: [Use lodash chain for better readability](https://github.com/frakt-solana/banx-ui/pull/381/commits/91428c84d271149e96d516cd11ce86ed8b1d9608)

File: src/pages/OffersPage/components/OffersTabContent/components/ActiveLoansTable/helpers.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+export const findBestOffer: FindBestOffer = ({ loan, offers }) => {
+  const offersByMarket = offers.filter((offer) => offer.hadoMarket === loan.fraktBond.hadoMarket)
+
+  const filteredOffers = offersByMarket.filter(
+    (offer) => calculateLoanValue(offer) > calculateLoanRepayValue(loan),
+  )
+
+  const sortedOffers = sortBy(filteredOffers, (offer) => offer.fundsSolOrTokenBalance)
+
+  return sortedOffers[0]
```
[the remark this answers is not in the document]
Comment (iamsphere) [reply src/pages/OffersPage/components/OffersTabContent/components/ActiveLoansTable/helpers.ts]: f32a2bb52febd05fd68d65e464e95b13320ff715

File: src/pages/OffersPage/components/OffersTabContent/components/ActiveLoansTable/helpers.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  const hasRefinanceOffers = findBestOffer({ loan, offers })
+  const isLoanUnderWater = isUnderWaterLoan(loan)
+
+  return !isLoanExpired && !isTerminatingStatus && !hasRefinanceOffers && isLoanUnderWater
+}
+
+type FindBestOffer = (props: { loan: Loan; offers: Offer[] }) => Offer
+
+export const findBestOffer: FindBestOffer = ({ loan, offers }) => {
+  const offersByMarket = offers.filter((offer) => offer.hadoMarket === loan.fraktBond.hadoMarket)
```
[the remark this answers is not in the document]
Comment (iamsphere) [reply src/pages/OffersPage/components/OffersTabContent/components/ActiveLoansTable/helpers.ts]: f32a2bb52febd05fd68d65e464e95b13320ff715

File: src/pages/OffersPage/hooks.ts [hunk 6]
```diff
@@ -3,41 +3,112 @@ import { useEffect, useMemo } from 'react'
...
+            existingLoan.loan.publicKey === loan.publicKey
+              ? convertLoanToOptimistic(loan, walletPublicKey)
+              : existingLoan,
+          )
+        }),
+      )
+  },
+}))
+
+export const USE_USER_OFFERS_V2_QUERY_KEY = 'userOffersV2'
```
[the remark this answers is not in the document]
Comment (iamsphere) [reply src/pages/OffersPage/hooks.ts]: 4cbd016b210986ae1567d93e4afda8fb62fdafef

File: src/pages/OffersPage/components/OffersTabContent/components/OfferCard/hooks.ts [hunk 7]
```diff
@@ -1,69 +1,23 @@
...
-      >
-        Remove
-      </Button>
-    </div>
-  )
-}
+import { enqueueSnackbar } from '@banx/utils'
 
-const useActionsCell = (offer: TableUserOfferData) => {
+//TODO: rename useActionsCell
```
[the remark this answers is not in the document]
Comment (iamsphere) [reply src/pages/OffersPage/components/OffersTabContent/components/OfferCard/hooks.ts]: [Rename "useActionsCell" => "useOfferActions"](https://github.com/frakt-solana/banx-ui/pull/381/commits/c34ba0252b00e0edd73ca5fa54c286c0d14ec451)

File: src/pages/OffersPage/components/OffersTabContent/components/ActiveLoansTable/TableCells/InterestCell.tsx [hunk 8]
```diff
@@ -0,0 +1,41 @@
+import { FC } from 'react'
+
+import { createPercentValueJSX, createSolValueJSX } from '@banx/components/TableComponents'
+
+import { Loan } from '@banx/api/core'
+import { HealthColorIncreasing, formatDecimal, getColorByPercent } from '@banx/utils'
+
+import { calculateClaimValue } from '../../OfferCard/helpers'
```
[the remark this answers is not in the document]
Comment (iamsphere) [reply src/pages/OffersPage/components/OffersTabContent/components/ActiveLoansTable/TableCells/InterestCell.tsx]: [Move helpers to index](https://github.com/frakt-solana/banx-ui/pull/381/commits/95643099a270db7d3842d9703d2ecc580cecc4d5)

File: src/pages/OffersPage/components/OffersTabContent/components/ActiveLoansTable/ActiveLoansTable.module.less (hunk 1 above)
[the remark this answers is not in the document]
Comment (sablevsky) [reply src/pages/OffersPage/components/OffersTabContent/components/ActiveLoansTable/ActiveLoansTable.module.less:13]: ок

**Pull request merged**